core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java

```diff
@@ -411,6 +411,33 @@ public String toString() {
     });
   }
 
+  private UfsStatus[] filterInvalidPaths(UfsStatus[] statuses, String listedPath) {
+    // This is a temporary fix to prevent us from choking on paths containing '?'.
+    if (statuses == null) {
+      return null;
+    }
+    int removed = 0;
+    for (UfsStatus status : statuses) {
+      if (status.getName().contains("?")) {
+        LOG.warn("Ignoring {} while listing {} since it contains '?'", status.getName(),
+            listedPath);
+        removed++;
+      }
+    }
+    if (removed > 0) {
+      UfsStatus[] newStatuses = new UfsStatus[statuses.length - removed];
+      int i = 0;
+      // We perform two passes to keep the common case (no invalid names) very cheap.
+      for (UfsStatus status : statuses) {
+        if (!status.getName().contains("?")) {
```

Comment:
We can probably just null the entry to avoid the second `contains` check which I think is the expensive operation.